Classes/Emogrifier.php

```diff
@@ -397,6 +420,57 @@ public function removeUnprocessableHtmlTag($tagName)
     }
 
     /**
+     * Normalizes the value of the "style" attribute and saves it.
+     *
+     * @param \DOMElement $node
+     *
+     * @return void
+     */
+    private function saveOriginalStyle(\DOMElement $node)
+    {
+        $normalizedOriginalStyle = preg_replace_callback(
+            '/[A-z\\-]+(?=\\:)/S',
+            function (array $m) {
+                return strtolower($m[0]);
+            },
+            $node->getAttribute('style')
+        );
+
+        // in order to not overwrite existing style attributes in the HTML, we
+        // have to save the original HTML styles
+        $nodePath = $node->getNodePath();
+        if (!isset($this->styleAttributesForNodes[$nodePath])) {
+            $this->styleAttributesForNodes[$nodePath] = $this->parseCssDeclarationBlock(
+                $normalizedOriginalStyle
+            );
+            $this->visitedNodes[$nodePath] = $node;
+        }
+
+        $node->setAttribute('style', $normalizedOriginalStyle);
+    }
+
+    /**
+     * Normalizes the style value for all tags that originally contained a
+     * style attribute.
+     *
+     * return @void
+     */
+    private function normalizeOriginalStyles()
```

Comment:
So we should have more unit tests (a.k.a. "documentation tests") that test what functionality still remains even if the options are switched off.